File: problems/13/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    var mapping = {
    ["IV"]: 4,
    ["IX"]: 9,
    ["I"]: 1,
    ["V"]: 5,
    ["XC"]: 90,
    ["XL"]: 40,
    ["X"]: 10,
    ["L"]: 50,
    ["CM"]: 900,
    ["CD"]: 400,
    ["C"]: 100,
    ["D"]: 500,
    ["M"]: 1000
  };
  var keys = Object.keys(mapping);
  var sum = 0;

  while (s.length > 0) {
      keys.forEach(key => {
          if (s.startsWith(key)) {
              sum += mapping[key];
              s = s.substring(key.length);
          }
      });
  }

  return sum;
};

File: problems/13/solution.cs
public class Solution {
    public int RomanToInt(string s) {
        var mapping = new Dictionary<char, int> {
            { 'I', 1 },
            { 'V', 5 },
            { 'X', 10 },
            { 'L', 50 },
            { 'C', 100 },
            { 'D', 500 },
            { 'M', 1000 }
        };
        var sum = 0;
        
        for (var i = 0; i < s.Length; i++) {
            if (i != 0 && mapping[s[i - 1]] < mapping[s[i]]) {
                sum -= 2 * mapping[s[i - 1]];
            }
            
            sum += mapping[s[i]];
        }
        
        return sum;
    }
}